### DISCUSSION TITLE
How to handle bottle :unneeded deprecation for external taps

### DISCUSSION THREAD

**Discussion by cippaciong** [Homebrew/discussions#2311] [Writing Formulae/Casks] [answered]
Hello, when I ran `brew upgrade` this morning, a bunch of warning messages popped up about `bottle :unneeded` being deprecated
```
Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the derailed/k9s tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/derailed/homebrew-k9s/Formula/k9s.rb:9
```

All the warnings came from external taps and tracing back the issue (https://github.com/Homebrew/homebrew-core/issues/75943 https://github.com/Homebrew/brew/pull/11239) it is clear why the option has been deprecated for Homebrew/core.

My question is: how should the external tap providers handle this deprecation? Is it enough to remove the option and formulae will be built locally as usual? Are they required to set up a bottle repository like Homebrew/core did?

**Comment by SMillerDev** [comment]
Tagging in @Homebrew/brew

**Comment by fxcoudert** [comment]
It's not clear from the two PRs: here https://github.com/Homebrew/brew/pull/11622 and there https://github.com/Homebrew/brew/pull/12259 

I think the recommended use outside of homebrew-core is simply not to have a `bottle` directive.

**Comment by carlocab** [comment]
> Is it enough to remove the option and formulae will be built locally as usual?

I believe this should be enough. `bottle :unneeded` is a no-op outside of Homebrew/core anyway.

**Comment by MikeMcQuaid** [comment, marked as answer]
> Is it enough to remove the option and formulae will be built locally as usual?

Yes 👍🏻 

> Are they required to set up a bottle repository like Homebrew/core did?

No.

**Comment by cippaciong** [comment]
Thanks to all of you for your replies 🙂
